corehq/apps/commtrack/processing.py

```diff
@@ -154,3 +188,54 @@ def process_stock(xform, case_db=None):
         relevant_cases=relevant_cases,
         stock_report_helpers=stock_report_helpers,
     )
+
+
+def _adjust_ledger_values(original_balance, stock_transaction):
+    if stock_transaction.report.type == stockconst.REPORT_TYPE_BALANCE:
+        quantity = stock_transaction.stock_on_hand
+    elif stock_transaction.report.type == stockconst.REPORT_TYPE_TRANSFER:
+        quantity = stock_transaction.quantity
+    else:
+        raise ValueError()
+
+    ledger_values = compute_ledger_values(
+        lambda: original_balance, stock_transaction.report.type, quantity)
+
+    if stock_transaction.report.type == stockconst.REPORT_TYPE_BALANCE:
+        assert stock_transaction.stock_on_hand == ledger_values.balance
+        # this is currently always 0 for some reason
+        stock_transaction.quantity = ledger_values.delta
+    elif stock_transaction.report.type == stockconst.REPORT_TYPE_TRANSFER:
+        stock_transaction.stock_on_hand = ledger_values.balance
+        assert stock_transaction.quantity == ledger_values.delta, (stock_transaction.quantity, ledger_values.delta)
+    else:
+        raise ValueError()
```

Comment:
Yeah, I like to do this, because even though right now it's unreachable, code gets moved around a lot; if this gets disconnected from the preceding code in a meaningful way, I'd want this check to be there.